File: pclassic/2022_Fall/salaj/q5.py
def adj(start, dest):
    if start == "A" or start =="C":
        return dest == "B" or dest == "D"
    else:
        return dest =="A" or dest =="C"

def timefor(n, k, order, duration, person):
    start = person[0]
    end = person[1]
    time = int(person[2])

    stff = "".join([order[x]*duration[x] for x in range(n)])
    adjfound = adj(start, end)

    for i in range(time, len(stff)):
        adjfound = adjfound or adj(start, stff[i])
        if adjfound and stff[i] == end:
            return i-time
    return -1


# Implement the following function
def evacuate(n, k, order, duration, sfx):
    res = []
    for person in sfx:
        res.append(timefor(n, k, order, duration, person))
    return res


tests = int(input().strip()) # Number of test cases
for test in range(tests):
    line = input().strip().split(' ')
    n = int(line[0])
    k = int(line[1])
    order = input().strip().split(' ') # List of which portals turn on in order
    duration = input().strip().split(' ') # List of times when portals are on for
    duration = [int(x) for x in duration]
    sfx = [] # List of triplets (start_portal, destination_portal, arrival_time)
    for _ in range(k):
        sfx.append(input().strip().split(' '))
    ret_arr = evacuate(n, k, order, duration, sfx)
    ret_arr = [str(x) for x in ret_arr]
    print('\n'.join(ret_arr))

File: pclassic/2022_Fall/q5.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

/* Return the shortest for a person to get to the finish destination */
vector<int> evacuate(int n, int k, string order[], int duration[], string sfx[][3]) {
    vector<int> v;
    return v;
}



int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int T, n, k;
    cin >> T;
    while(T--)
    {
        cin >> n >> k;
        string order[n]; //The order in which the portals turn on
        int duration[n]; //The time in which each portal stays on
        string sfx[k][3];
        for(int i = 0; i < n; i++)
        {
            cin >> order[i];
        }
        
        for (int i = 0; i < n; i++) cin >> duration[i];
        
        for (int i = 0; i < k; i++)
        {
            cin >> sfx[i][0] >> sfx[i][1] >> sfx[i][2];
        }
        
        vector<int> res = evacuate(n, k, order, duration, sfx);
        
        for (int i = 0; i < res.size(); i++) 
        {
            cout << res[i] << endl;
        }
    }
}